lib/Dialect/RTL/Ops.cpp

```diff
@@ -697,6 +738,20 @@ void MulOp::getCanonicalizationPatterns(OwningRewritePatternList &results,
 
       APInt value, value2;
 
+      // mul(x, c) -> mul(shl(x, n)), where c is a power of two, & n = log2(c)
+      if (size == 2 && matchPattern(inputs.back(), m_RConstant(value)) &&
+          value.isPowerOf2()) {
+        auto shift =
+            rewriter.create<ConstantOp>(op.getLoc(), value.exactLogBase2(),
```

Comment:
Does it work if you drop passing the integer type explicitly?  I think rtl::ConstantOp has a ctor that takes an `APInt` alone.

Reply:
Why not just shift x?

Reply:
Dropping the type says theres no matching member function for the call to `build`. 
Can you explain what you mean Stephen?

Reply:
Ok oh, I thought `exactLogBase2` returned an APInt.  Plz disregard!

Reply:
@cgyurgyik If c is a power of two then mul(x, c) -> shl(x, log2(c)).  I think the code is correct but the comment has an extra mul().